src/main/java/io/github/benas/jpopulator/impl/PopulatorImpl.java

```diff
@@ -139,9 +164,24 @@ public PopulatorImpl() {
             throw new IllegalArgumentException("The number of beans to populate must be positive.");
         }
         List<Object> beans = new ArrayList<Object>();
-        for (int i = 0; i < size; i++) {
-            Object bean = populateBean(type, excludedFields);
-            beans.add(bean);
+        if (isSupportedType(type)) {
+            BeanFieldHolder beanFieldHolder = new BeanFieldHolder();
```

Comment:
The code needs an instance of java.lang.reflect.Field of type Object to pass into populateSupportedType at line 177, and I don't know how to construct one. 

I agree the code is a bit sloppy here. 
- There is no need to instantiate beanFieldHolder which is never used
- A  constant, say OBJECT_FIELD, could be used in place of the local variable field.
- The name ObjectFieldSupplier would better match its use than BeanFieldHolder. 

How would you like me to proceed?